### DISCUSSION TITLE
CS8795 Partial method must have an implementation part because it has accessibility modifiers.

### DISCUSSION THREAD

**Discussion by CrossShot** [riok/mapperly#519] [Q&A] [answered]
I am trying to implement mapperly into my project and I am receiving the referenced error when trying to make a partial method.

I even did a fresh pull of the main branch and made no changes, and the error appears in the sample project as well.

What am I doing wrong?

![image](https://github.com/riok/mapperly/assets/40874511/cfb97ddd-d1c1-47bd-abfe-7848715e8e58)

**Comment by TimothyMakkison** [comment]
Restarting vs sometimes helps, are there also warnings for the integration tests?

Running `dotnet test` might give a different error. I did find that VS gave me an error about referencing Mapperly.Abstractions

**Reply by CrossShot** [reply]
So, restarting Visual Studio on my home machine that is running VS 2022 & .NET 7 did fix the red squiggle for me and I was able to run it.

However, on the project I'm currently working on I'm locked into VS 2019 & .NET 5 and unfortunately it still throws that error after a restart.

**Comment by CrossShot** [comment, marked as answer]
It seems the issue is that because I'm locked into .NET 5 and C# 7.3 I'm unable to use mapperly for my project.

I need the [Extended Partial Methods](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/extending-partial-methods) feature introduced in C# 9.0.

Thanks for the help @TimothyMakkison

**Comment by DermotMurphy** [comment]
I am experiencing a similar (the same) issue on building in Visual Studio 2022, but I am running .Net7

After upgrading to v3.0.0, I exited Visual Studio and deleted the .vs folder but get error messages like: 
"Partial method 'SupplierExtensions.ToVm(SupplierListResponse)' must have an implementation part because it has accessibility modifiers."

I can build without errors at command line using command "dotnet build".

Very sadly rolling back the changes

**Reply by latonz** [reply]
Does the error still occur if you close VS, run `dotnet clean` in a terminal and start VS again?

**Comment by DermotMurphy** [comment]
Good idea, I'll try it tomorrow morning 👍

**Comment by DermotMurphy** [comment]
Following Lars suggestion about upgrading Mapperly to version V3.0

1. In Visual Studio, Use Package Manager to update 3 projects
2. Close Visual Studio
3. Clean obj & bin folders and delete the .vs folder
4. At the command line 'dotnet build' ... succeeds with no errors
5. Open Visual Studio & Rebuild solution
6. Sadly, many CS8795 errors

This **IS** very strange because the build at the command prompt succeeded but failed in Visual Studio 😪

**Reply by latonz** [reply]
Could probably also be related to https://github.com/riok/mapperly/issues/650. Are you using enum configurations (EnumMappingStrategy, IgnoreObsoleteMembersStrategy, MappingConversionType, PropertyNameMappingStrategy, ...)?

**Reply by DermotMurphy** [reply]
This is the start of a class ...

[Mapper(EnumMappingStrategy = EnumMappingStrategy.ByName)]

public static partial class SupplierMappingExtensions
{

**Reply by DermotMurphy** [reply]
Checking through jhe warning messages, there are three of these:
Warning	CS8785	Generator 'MapperGenerator' failed to generate source. It will not contribute to the output and compilation errors may occur as a result. Exception was of type 'InvalidOperationException' with message 'Could not resolve enum reflection type of EnumMappingStrategy or Riok.Mapperly.Abstractions.EnumMappingStrategy is not supported'	Insite.Server.Core.Application	E:\Development\@NewIntranet_2023-07-14_13.53.05\InsiteHosted\src\Server\Core\Insite.Server.Core.Application\CSC	1	N/A

**Reply by latonz** [reply]
Thank you for these details, this seems to be the same root cause as https://github.com/riok/mapperly/issues/650, which we are currently investigating.

**Reply by DermotMurphy** [reply]
OK, It seems to be an issue between Visual Studio and Mapperly.

Having read #650 , I have opened the solution In Rider 2023.1.2, performed a build **and** everything is OK, no warning messages and no errors 
Sadly, I can't just switch to using Rider as my colleagues do not use it ☹

**Reply by DermotMurphy** [reply]
BTW. I am using VS2022 Professional Version 17.7.0

**Reply by DermotMurphy** [reply]
This is such a great product. 

I really like it and have recommended it to my colleagues and friends. 

I really appreciate your effort in producing and maintaining it. 

Thanks you 👏👏👏

**Reply by latonz** [reply]
Mapperly 3.1.0 was just released which should fix this in VS 17.6 and later.

**Comment by DermotMurphy** [comment]
Fantastic... I'll get right on it come Monday morning.

Have a great weekend 😄

**Comment by DermotMurphy** [comment]
Hi Lars,

Sorry, I mean to do this work on Monday but things didn't turn out that way.

I have installed the latest version of Mapperly and it is now working fine
... Yay!

I also upgraded Visual Studio to :

Microsoft Visual Studio Professional 2022 (64-bit) - Current
Version 17.7.2

Happily Mapperly is functioning well ... another clap for you & your team.

👏👏👏👏
